force-app/main/default/classes/AffiliationMappingsService.cls

```diff
@@ -101,6 +101,47 @@ public virtual with sharing class AffiliationMappingsService {
         this.locateAffiliationMappingsMapper().updateAffiliationMappings(afflMappingModel);
     }
 
+    /*****************************************************************************
+     * @description Checks user permissions and update the auto enrollment mapppings.
+     *****************************************************************************/
+    public virtual void updateAutoEnrollmentMappings(
+        String mappingName,
+        String oldAccountRecordType,
+        String newAccountRecordType,
+        String autoProgramEnrollmentStatus,
+        String autoProgramEnrollmentRole
+    ) {
+        this.locateUserPermissionService().checkViewSetupForCurrentUser();
+        if (oldAccountRecordType != newAccountRecordType) {
```

Comment:
Consider using the more simple use case first and make it function as a guard clause instead of an if, else...